ClinGen/clincoded
# Connects to #1187. Connects to #1188. Connects to #1191.

**Steps to test #1187:**
1. Create a new Experimental evidence in GCI.
2. On the subsequent form page, select the _**"Biochemical Function"** Experiment type_ and then select either option A or B. Expect to see the **Select a GO term for this gene (view existing GO annotations for this gene in UniProt or search for a new term using QuickGO). The GO term must be from the GO aspect "Molecular Function" or "Biological Process."** description above the _GO ID_ text field. And clicking on the **existing GO annotations for this gene** link (in the description above) should open the expected UniProt page with the associated gene in a new window.
![image](https://cloud.githubusercontent.com/assets/6956310/21740706/8e72d6d6-d476-11e6-9994-d82f5518e5ac.png)
3. Now select the _**"Functional Alteration"** Experiment type_ and also expect to see the same behavior as descriobed above.

**Steps to test #1188:**
1. Create a new Family evidence in GCI.
2. On the subsequent form page, fill the _required_ fields and add a Proband individual with associated variant(s). Save the page and proceed to the Proband form page.
3. On the Proband individual form page, expect to see a note **"Note: a variant's gene impact must be specified in order to score this proband."** and a link **"Curate variant's gene impact"** appended to each associated variant. In addition, a link "**View variant evidence in Variant Curation Interface"** is appended to each variant regardless a Proband or not.
![image](https://cloud.githubusercontent.com/assets/6956310/21741028/7c8c8242-d47f-11e6-8d62-4f5fc0de4efc.png)
4. On the Proband individual form page, expect to see a note in yellow box being shown in the Proband Score panel.
![image](https://cloud.githubusercontent.com/assets/6956310/21744294/60bb8cdc-d4c7-11e6-8bd5-ee015cf4b7be.png)
5. Add a non-proband individual to the same Family. On this individual's form page, expect to see **NO** note or the "Curate variant's gene impact" link being shown for the associated variants.

**Steps to test #1191:**
1. Create a new Proband individual evidence in GCI and score it with a **"0"** custom score (different from the default score).
2. Try saving the page without providing an explanation. You should expect to see a warning being shown and the form not successfully submitted.
3. Provide an explanation in the textarea field and try saving the page again. You should expect to see the form being submitted successfully and the user being returned to Curation Central.
4. Return to this Proband individual's form page and expect to see the custom score of **"0"** being shown correctly in the Proband Score panel.
![image](https://cloud.githubusercontent.com/assets/6956310/21744320/d9ef27b2-d4c7-11e6-95e0-de5e9e50954d.png).
5. Create a new Case-Control evidence in GCI and select **"0"** as the score. Save the page and then return to the form page again. Expect the **"0"** score being shown correctly in the Case-Control Score panel.
![image](https://cloud.githubusercontent.com/assets/6956310/21740735/37a48fa6-d477-11e6-8e92-674331348744.png)
